src/document.rs

```diff
@@ -1,15 +1,71 @@
 //! Abstractions for CMark Documents and Howser Templates.
 
+extern crate env_logger;
 extern crate regex;
 extern crate unicode_segmentation;
 
 use self::regex::Regex;
-use std::collections::HashMap;
-use std::cell::RefCell;
-use errors::{HowserError, HowserResult};
+use self::unicode_segmentation::UnicodeSegmentation;
+use constants::{DITTO_TOKEN, MANDATORY_PROMPT, OPTIONAL_PROMPT, PROMPT_PATTERN};
+use data::ElementType;
 use data::{MatchType, NodeData};
-use doogie::Node;
 use doogie::constants::*;
+use doogie::Node;
+use errors::{HowserError, HowserResult, SpecWarning};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use validator::types_match;
+
+#[derive(Debug)]
+enum LookaheadType {
+    IntegratedLiteral(Node),
+    IntegratedOccupied((Node, MatchType)),
```

Comment:
Why the embedded tuple here? (and below) `IntegratedOccupied(Node, MatchType)` seems more natural.

Reply:
Because I was silly and forgot that enums take more than one internal value.

Fixed